examples not work when republishing 2 medias

https://github.com/bluenviron/gortsplib/blob/main/examples/client-play-to-record/main.go
first try nil pointer dereference
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x104997f44]

goroutine 25 [running]:
github.com/bluenviron/gortsplib/v4.(*clientFormat).writePacketRTP(0x0, 0x14000304000, {0x14000304004?, 0x30000007d?, 0x104cd9760?})
        /Users/a/go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.16.0/client_format.go:188 +0x34
github.com/bluenviron/gortsplib/v4.(*Client).WritePacketRTPWithNTP(0x14000151088, 0x1400002c2a0, 0x14000304000, {0x10479f4e4?, 0x14000286de8?, 0x104cd9760?})
        /Users/a/go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.16.0/client.go:2258 +0x11c
github.com/bluenviron/gortsplib/v4.(*Client).WritePacketRTP(0x14000151088, 0x1400002c2a0, 0x14000304000)
        /Users/a/go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.16.0/client.go:2237 +0x4c
main.main.func1(0x14000286df8?, {0x1047fc900?, 0x14000286e08?}, 0x14000286db0?)
        /Users/a/work/hgmDemo/hdgCpoDemo/hdgCpoDemo/make/tRtspCopy2/main.go:64 +0x3c
github.com/bluenviron/gortsplib/v4.(*Client).OnPacketRTPAny.func1(0x140000b40c0?)
        /Users/a/go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.16.0/client.go:2206 +0x38
github.com/bluenviron/gortsplib/v4.(*clientFormat).handlePacketRTP(0x140001558f0, 0x14000304000, {0x7bbcce1f416ee0e?, 0x14000304004?, 0x104cd9760?})
        /Users/a/go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.16.0/client_format.go:179 +0x108
github.com/bluenviron/gortsplib/v4.(*clientFormat).readPacketRTPUDP(0x140001558f0, 0x14000115530?)
        /Users/a/go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.16.0/client_format.go:154 +0xcc
github.com/bluenviron/gortsplib/v4.(*clientMedia).readPacketRTPUDPPlay(0x1400016a2d0, {0x1400022ec00?, 0x14000286f28?, 0x10499b6cc?})
        /Users/a/go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.16.0/client_media.go:369 +0x1cc
github.com/bluenviron/gortsplib/v4.(*clientUDPListener).run(0x1400016a3f0)
        /Users/a/go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.16.0/client_udp_listener.go:136 +0x174
created by github.com/bluenviron/gortsplib/v4.(*clientUDPListener).start in goroutine 35
        /Users/a/go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.16.0/client_udp_listener.go:93 +0xbc
panic: exit status 2
```
second try, change code to : got destroyed: session timed out
```
	// read RTP packets from the reader and route them to the publisher
	reader.OnPacketRTPAny(func(media *description.Media, _ format.Format, pkt *rtp.Packet) {
		err2 := publisher.WritePacketRTP(media, pkt)
		if err2 != nil {
			log.Printf("ERR: %v", err2)
		}
	})
```
```
2025/07/30 17:08:56 INF [RTSP] [session 6dd8b3c5] is publishing to path 'xxxxx8', 2 tracks (H264, MPEG-4 Audio)
2025/07/30 17:09:06 INF [RTSP] [conn xxxx:53510] closed: terminated
2025/07/30 17:09:06 INF [path xxxxx8] runOnReady command stopped
2025/07/30 17:09:06 INF [path xxxxx8] runOnNotReady command launched
2025/07/30 17:09:06 INF [RTSP] [session 6dd8b3c5] destroyed: session timed out
```
third try, change code to : worked.
```
	// connect to the server and start recording the same medias
	publisher := gortsplib.Client{}
	tcp:=gortsplib.TransportTCP
	publisher.Transport = &tcp
	err = publisher.StartRecording("rtsp://xxxxx/xxxxx8", desc)
	if err != nil {
		panic(err)
	}
	defer publisher.Close()

	// read RTP packets from the reader and route them to the publisher
	reader.OnPacketRTPAny(func(media *description.Media, _ format.Format, pkt *rtp.Packet) {
		err2 := publisher.WritePacketRTP(media, pkt)
		if err2 != nil {
			log.Printf("ERR: %v", err2)
		}
	})
```
so I can not publish rtsp with udp. only tcp worked.

--- a/examples/client-play-to-record/main.go
+++ b/examples/client-play-to-record/main.go
@@ -59,8 +59,8 @@ func main() {
 	defer publisher.Close()
 
 	// read RTP packets from the reader and route them to the publisher
-	reader.OnPacketRTPAny(func(_ *description.Media, _ format.Format, pkt *rtp.Packet) {
-		err2 := publisher.WritePacketRTP(desc.Medias[0], pkt)
+	reader.OnPacketRTPAny(func(media *description.Media, _ format.Format, pkt *rtp.Packet) {
+		err2 := publisher.WritePacketRTP(media, pkt)
 		if err2 != nil {
 			log.Printf("ERR: %v", err2)
 		}

